### DISCUSSION TITLE
Can we get UTF-8 support for superbooga?

### DISCUSSION THREAD

**Discussion by caterpillarpants** [oobabooga/text-generation-webui#4523] [Ideas]
Can we get UTF-8 support for superbooga? Is it a pain? I don't know much about it, but it seems a bit weird.

weirdness:
take the word "rama" in Sanskrit this is "रम" which is UTF-8: e0a4b0,e0a4ae
but the word "rame" in Sanskrit would be "रमॆ" which is UTF-8: e0a4b0,e0a4ae,e0a586
see. not so straightforward. so...
ra or र is e0a4b0
ma or म is e0a4ae
e or ॆ is e0a586

but when you put the three together you don't get ramae instead you get rame or रामॆ as the second "a" is removed and replaced with the "e".

**Comment by berkut1** [comment]
But UTF is supported. What you are describing is probably a tokenization issue (or a model issue) and you should contact the repository of whatever module you are using, superbooga is basically a web GUI only.

**Reply by caterpillarpants** [reply]
I wasn't clear. it's true I have a tokenizer problem, but that is really separate from why I'm asking for utf-8 support. You see, when I put in a pdf with utf-8 in superbooga, it is rejected. In the gui it says "ERROR" and in the terminal it complains and ends with...
`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe2 in position 10: invalid continuation byte`

or some variation. here is another one from a different pdf.

`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc7 in position 10: invalid continuation byte`

**Comment by caterpillarpants** [comment]
Oh! I get it! I'll bother the super project on it. Sorry! ha!

**Reply by TomLucidor** [reply]
Same problem is gonna happen with PrivateGPT so there has to be a common solution for text formatting
